tests/src/main/java/com/hazelcast/stabilizer/tests/map/GrowingMapTest.java

```diff
@@ -69,84 +68,83 @@ public void verify() throws Exception {
 
-                    long key = idGenerator.newId();
-                    long value = random.nextLong();
-                    keyIndex = k;
-                    keys[keyIndex] = key;
-                    values[keyIndex] = value;
+                insertIteration++;
+                if (insertIteration % logFrequency == 0) {
+                    log.info(Thread.currentThread().getName() + " At insert iteration: " + insertIteration);
+                }
+            }
 
-                    if (usePut) {
-                        map.put(key, value);
-                    } else {
-                        map.set(key, value);
-                    }
+            log.info(Thread.currentThread().getName() + " Inserted " + insertIteration + " key/value pairs");
+            return insertIteration;
+        }
 
-                    insertIteration++;
-                    if (insertIteration % logFrequency == 0) {
-                        log.info(Thread.currentThread().getName() + " At insert iteration: " + insertIteration);
-                    }
-                }
+        private void read(int keyIndex) {
+            int readIteration = 0;
+            while (!testContext.isStopped() && readIteration < keyIndex) {
+                long key = keys[readIteration];
+                long value = values[readIteration];
 
-                if (readValidation) {
-                    for (int k = 0; k <= keyIndex; k++) {
-                        if (testContext.isStopped()) {
-                            break;
-                        }
-
-                        long key = keys[k];
-                        long value = values[k];
-
-                        long found = map.get(key);
-                        if (found != value) {
-                            throw new RuntimeException("Unexpected value found");
-                        }
-
-                        readIteration++;
-                        if (readIteration % logFrequency == 0) {
-                            log.info(Thread.currentThread().getName() + " At read iteration: " + readIteration);
-                        }
-                    }
+                long found = map.get(key);
+                if (found != value) {
+                    throw new RuntimeException("Unexpected value found");
                 }
 
-                for (int k = 0; k <= keyIndex; k++) {
-                    if (testContext.isStopped() && !removeOnStop) {
-                        break;
-                    }
+                readIteration++;
+                if (readIteration % logFrequency == 0) {
+                    log.info(Thread.currentThread().getName() + " At read iteration: " + readIteration);
+                }
+            }
+        }
 
-                    long key = keys[k];
-                    long value = values[k];
+        private void delete(int keyIndex) {
+            int deleteIteration = 0;
+            while ((!testContext.isStopped() || removeOnStop) && deleteIteration < keyIndex) {
```

Comment:
Should work out. The test may fail in validate stage if `removeOnStop = false` and the delete phase is not completed. But per default it's `true`, so the delete phase will always complete.